### PR TITLE
fix: disallow target account when creating position for integrity pool

### PR DESCRIPTION
**Pull request by keyvankhademi**

### CODE REVIEW

File: staking/programs/staking/src/error.rs [hunk 1]
```diff
@@ -82,6 +82,8 @@ pub enum ErrorCode {
     MissingTargetAccount,
     #[msg("The slash ratio should be between 0 and 1")] // 6038
     InvalidSlashRatio,
-    #[msg("Other")] //6039
+    #[msg("The target account is not allowed for this operation")] // 6039
```
Comment (guibescos) [inline staking/programs/staking/src/error.rs:85]: The target account is only expected when dealing with the governance target

Comment (guibescos) [inline staking/programs/staking/src/error.rs:85]:
```suggestion
    #[msg("The target account is only expected when dealing with the governance target")] // 6039
```

File: staking/programs/staking/src/lib.rs [hunk 2]
```diff
@@ -189,6 +189,10 @@ pub mod staking {
         }
 
         if let TargetWithParameters::IntegrityPool { .. } = target_with_parameters {
+            require!(
```
Comment (guibescos) [inline staking/programs/staking/src/lib.rs:192]: need to update `close_position` as well

Comment (keyvankhademi) [reply staking/programs/staking/src/lib.rs:192]: done

File: staking/integration-tests/tests/pool_authority.rs [hunk 3]
```diff
@@ -128,6 +129,76 @@ fn test_pool_authority() {
...
+        Some(&payer.pubkey()),
+        &[&payer, &pool_authority],
+        svm.latest_blockhash(),
+    );
+
+    assert_anchor_program_error!(
+        svm.send_transaction(create_position_tx),
+        ErrorCode::UnexpectedTargetAccount,
+        0
+    );
```
Comment (guibescos) [inline staking/integration-tests/tests/pool_authority.rs:143]: this test is duplicated below

Comment (keyvankhademi) [reply staking/integration-tests/tests/pool_authority.rs:143]: the name was wrong, it is actually testing close position. Fixed.

**Pull request merged**